src/BenchmarkDotNet.Core/Toolchains/DotNetCli/DotNetCliGenerator.cs

```diff
@@ -91,7 +96,11 @@ protected override void Cleanup(ArtifactsPaths artifactsPaths)
                 }
                 catch (Exception) when (attempt++ < 5)
                 {
+#if UAP
```

Comment:
I think whole Toolchains namespace should be excluded for UAP. There will be no support for running BDN as host.